Added Helm3 install integration tests

nrjpoddar opened the pull request:
[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure


Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

istio-testing commented:
@nrjpoddar: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
release-notes_istio | fbbaa54df6bdd9e386f0b1a3efee6955381ea9c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/release-notes_istio/1321607396686041088) | `/test release-notes_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
release-notes_istio | fbbaa54df6bdd9e386f0b1a3efee6955381ea9c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/release-notes_istio/1321607396686041088) | `/test release-notes_istio`
gencheck_istio | fbbaa54df6bdd9e386f0b1a3efee6955381ea9c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321607395838791680) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
release-notes_istio | fbbaa54df6bdd9e386f0b1a3efee6955381ea9c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/release-notes_istio/1321607396686041088) | `/test release-notes_istio`
gencheck_istio | fbbaa54df6bdd9e386f0b1a3efee6955381ea9c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321607395838791680) | `/test gencheck_istio`
lint_istio | fbbaa54df6bdd9e386f0b1a3efee6955381ea9c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321607395008319488) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | fbbaa54df6bdd9e386f0b1a3efee6955381ea9c6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321607395008319488) | `/test lint_istio`
gencheck_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321608158694608896) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321608158694608896) | `/test gencheck_istio`
lint_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321608157864136704) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321608158694608896) | `/test gencheck_istio`
lint_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321608157864136704) | `/test lint_istio`
analyze-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321608164566634496) | `/test analyze-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321608158694608896) | `/test gencheck_istio`
lint_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321608157864136704) | `/test lint_istio`
analyze-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321608164566634496) | `/test analyze-tests_istio`
unit-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/unit-tests_istio/1321608144438169600) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,218 @@
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name: IstioNamespace,
+		},
+	}, kubeApiMeta.CreateOptions{}); err != nil {
```
nit: you can use kerrors.isAlreadyExists() instead of another Get

nrjpoddar replied:
Done

nrjpoddar commented:
/test unit-tests_istio

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321608158694608896) | `/test gencheck_istio`
lint_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321608157864136704) | `/test lint_istio`
analyze-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321608164566634496) | `/test analyze-tests_istio`
unit-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/unit-tests_istio/1321615920887500800) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn reviewed (commented):
looks good!

nrjpoddar commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,218 @@
+	if err := h.DeleteChart(BaseReleaseName, IstioNamespace); err != nil {
+		t.Errorf("failed to delete %s release", BaseReleaseName)
+	}
+	if err := cs.CoreV1().Namespaces().Delete(context.TODO(), IstioNamespace, kubeApiMeta.DeleteOptions{}); err != nil {
```
Running tests in succession is causing intermittent failures due to the namespace terminating while the new Helm install happens. Is there a util to ensure a namespace is completely gone before proceeding with the next step?

jacob-delgado commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,218 @@
+// deleteIstio deletes installed Istio Helm charts and resources
+func deleteIstio(t *testing.T, cs resource.Cluster, h *helm.Helm) {
+	scopes.Framework.Infof("cleaning up resources")
+	if err := h.DeleteChart(IngressReleaseName, IstioNamespace); err != nil {
```
you should do egress first as it was the last one installed, then ingress

nrjpoddar replied:
Good catch, resolved!

jacob-delgado commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,218 @@
+	if err := h.DeleteChart(EgressReleaseName, IstioNamespace); err != nil {
+		t.Errorf("failed to delete %s release", EgressReleaseName)
+	}
+	if err := h.DeleteChart(BaseReleaseName, IstioNamespace); err != nil {
```
You are missing the istio chart itself.

nrjpoddar replied:
Good catch added

jacob-delgado commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,218 @@
+	}
+	if err := cs.CoreV1().Namespaces().Delete(context.TODO(), IstioNamespace, kubeApiMeta.DeleteOptions{}); err != nil {
+		t.Errorf("failed to delete %s namespace", IstioNamespace)
+	}
```
The CRDs remain after this test is done. You may wish to also remove them. I think the kind cluster is done on a per package basis.

nrjpoddar replied:
I left it behind intentional as that's what a user would do and Helm install/uninstall should work with existing CRDs.

jacob-delgado commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,218 @@
+			return fmt.Errorf("istio ingress gateway pod is not ready: %v", err)
+		}
+		if _, err := kubetest.CheckPodsAreReady(kubetest.NewSinglePodFetch(cs, IstioNamespace, "app=istio-egressgateway")); err != nil {
+			return fmt.Errorf("istio egress gateway  pod is not ready: %v", err)
```
nit: extra space between gateway and pod

nrjpoddar replied:
Addressed.

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321608157864136704) | `/test lint_istio`
analyze-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321608164566634496) | `/test analyze-tests_istio`
unit-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/unit-tests_istio/1321615920887500800) | `/test unit-tests_istio`
gencheck_istio | 22bbcf28b32d19e2fb0363de76a62daf557e7e99 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321848272590999552) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321608157864136704) | `/test lint_istio`
analyze-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321608164566634496) | `/test analyze-tests_istio`
unit-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/unit-tests_istio/1321615920887500800) | `/test unit-tests_istio`
gencheck_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321849028584935424) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
analyze-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321608164566634496) | `/test analyze-tests_istio`
unit-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/unit-tests_istio/1321615920887500800) | `/test unit-tests_istio`
gencheck_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321849028584935424) | `/test gencheck_istio`
lint_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321849036122099712) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
analyze-tests_istio | a6cee4af12486a45d574348978b3099daf994100 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321608164566634496) | `/test analyze-tests_istio`
gencheck_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321849028584935424) | `/test gencheck_istio`
lint_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321849036122099712) | `/test lint_istio`
unit-tests_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/unit-tests_istio/1321849014311718912) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

nrjpoddar commented:
I'm not sure why unit tests and all checks are failing as I haven't changed them. Can anyone help?

istio-testing commented:
@nrjpoddar: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/gencheck_istio/1321849028584935424) | `/test gencheck_istio`
lint_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/lint_istio/1321849036122099712) | `/test lint_istio`
unit-tests_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/unit-tests_istio/1321849014311718912) | `/test unit-tests_istio`
analyze-tests_istio | ffe6bba51781b47d2af8b35eafeff4f5aa19bad5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321849035287433216) | `/test analyze-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
analyze-tests_istio | 30879f47259d2274d7c22c93997a91553ca0f23d | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/analyze-tests_istio/1321874808639590400) | `/test analyze-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@nrjpoddar: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-multicluster-k8s-tests_istio | 5340db424452cf2891c8941e747e213968b5a368 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28385/integ-multicluster-k8s-tests_istio/1321918386849976320) | `/test integ-multicluster-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

nrjpoddar commented:
/test integ-multicluster-k8s-tests_istio

sdake commented:
@nrjpoddar the unit tests were or are busted because of a problem with Envoy that should be fixed shortly or has been fixed already.

Really nice work!

sdake commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,220 @@
+	"testing"
+	"time"
+
+	kubeapicore "k8s.io/api/core/v1"
```
super nit - typically these are v1 and metav1.

nrjpoddar replied:
Done

sdake commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,220 @@
+	ControlChartsDir    = "istio-control"
+	GatewayChartsDir    = "gateways"
+	retryDelay          = 2 * time.Second
+	retryTimeOut        = 20 * time.Minute
```
this timer is too long. The CI has a 10 minute timer on no output. Typically we would run this timer at 300 seconds.

nrjpoddar replied:
Done

sdake commented on tests/integration/helm/install_test.go:
```diff
@@ -0,0 +1,220 @@
+			if err != nil {
+				t.Fatal(err)
+			}
+			overrideValuesStr := `
```
nit - do quotes not work here?

nrjpoddar replied:
Not really for multi line use case

sdake commented on tests/util/sanitycheck/sanity_check.go:
```diff
@@ -0,0 +1,58 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sanitycheck
```
nice 👍

sdake reviewed (commented):
LGTM - I was not first to review, so I won't approve until the primary reviewer is good with the work.

Cheers,
-steve

nrjpoddar commented:
Thanks @sdake for the review.

I have addressed all the comments so if any of the reviewers can approve, I can merge and cherry-pick to release-1.8. Ping @howardjohn

nrjpoddar commented:
Corresponding test-infra PR https://github.com/istio/test-infra/pull/3021 to enable running in CI

istio-testing commented:
In response to a cherrypick label: new pull request created: #28431
